nyc/bills.py

```diff
@@ -161,133 +161,150 @@ def sponsorships(self, matter_id):
+
+        for topic in self.topics(matter_id) :
+            bill.add_subject(topic['MatterIndexName'].strip())
+
+        for relation in self.relations(matter_id):
             try:
-                for sponsorship in self.sponsorships(matter_id):
-                    bill.add_sponsorship(**sponsorship)
-            except KeyError:
-                version_errors.append(legistar_web)
-                continue
+                related_bill = self.endpoint('/matters/{0}', relation['MatterRelationMatterId'])
+            except scrapelib.HTTPError:
+                return
+            else:
+                date = related_bill['MatterIntroDate']
+                related_bill_session = self.session(self.toTime(date))
+                identifier = related_bill['MatterFile']
+                bill.add_related_bill(identifier=identifier,
+                                      legislative_session=related_bill_session,
+                                      relation_type='companion')
 
-            for attachment in self.attachments(matter_id):
+        try:
+            text = self.text(matter_id)
+        except KeyError:
+            self.version_errors.append(legistar_web)
+            return
 
-                if attachment['MatterAttachmentId'] == 103315:  # Duplicate
-                    continue
 
-                if attachment['MatterAttachmentName']:
-                    bill.add_document_link(attachment['MatterAttachmentName'],
-                                           attachment['MatterAttachmentHyperlink'],
-                                           media_type='application/pdf')
+        if text:
+            if text['MatterTextPlain']:
+                bill.extras['plain_text'] = text['MatterTextPlain'].replace(u'\u0000', '')
 
-            for topic in self.topics(matter_id) :
-                bill.add_subject(topic['MatterIndexName'].strip())
+            if text['MatterTextRtf']:
+                bill.extras['rtf_text'] = text['MatterTextRtf'].replace(u'\u0000', '')
 
-            for relation in self.relations(matter_id):
-                try:
-                    related_bill = self.endpoint('/matters/{0}', relation['MatterRelationMatterId'])
-                except scrapelib.HTTPError:
-                    continue
-                else:
-                    date = related_bill['MatterIntroDate']
-                    related_bill_session = self.session(self.toTime(date))
-                    identifier = related_bill['MatterFile']
-                    bill.add_related_bill(identifier=identifier,
-                                          legislative_session=related_bill_session,
-                                          relation_type='companion')
+        return bill
 
-            try:
-                text = self.text(matter_id)
-            except KeyError:
-                version_errors.append(legistar_web)
-                continue
+    def format_event(self, bill, action):
```

Comment:
+1 on breaking this into multiple methods. Let's find a better name than `format_event`